addons/mail/static/src/composer/composer.js

```diff
@@ -20,6 +21,10 @@ export class Composer extends Component {
             (focus) => {
                 if (focus && this.ref.el) {
                     this.ref.el.focus();
+                    if(this.caretLocationBeforeAddingEmoji) {
+                        this.ref.el.setSelectionRange(this.caretLocationBeforeAddingEmoji, this.caretLocationBeforeAddingEmoji);
+                        this.caretLocationBeforeAddingEmoji = null;
+                    }
```

Comment:
I was wondering why it was not just simply after `addEmoji` LOCs, but the current implementation of `useEffect` has side-effect to re-focus at the end of composer input...

`useEffect` is a bad smell equivalent to our `onChange` in the previous code of discuss. We should avoid it as much as possible.

I would suggest doing something to remove this `useEffect`. Make a test for coverage of chatter change of send message/log not in chatter, and reimplement a solution without this `useEffect`. If doing properly, then this "restore" of caret position should be applied immediately after `addEmoji`, not in a following `onChange`